app/src/main/java/dev/ukanth/ufirewall/log/LogRecyclerViewAdapter.java

```diff
@@ -107,42 +98,37 @@ private Bitmap getAppIcon26(PackageManager mPackageManager, ApplicationInfo appl
 
     @Override
     public void onBindViewHolder(ViewHolder holder, int position) {
-        data = logData.get(position);
+        LogData data = logData.get(position);
         PackageManager manager = context.getPackageManager();
-        holder.bind(logData.get(position),recyclerItemClickListener);
+        holder.bind(logData.get(position), recyclerItemClickListener);
         try {
-            HashMap<Integer,String> listMaps = Api.getPackagesForUser(Api.getListOfUids());
-            info = Api.getPackageDetails(context, listMaps, data.getUid());
-            if(info!=null && info.applicationInfo !=null) {
-                Object drawable = info.applicationInfo.loadIcon(manager);
-                if (drawable instanceof Bitmap)
-                    holder.icon.setImageBitmap((Bitmap)drawable);
-                else
-                    holder.icon.setBackground((Drawable)drawable);
-            } else{
+            HashMap<Integer, String> listMaps = Api.getPackagesForUser(Api.getListOfUids());
+            PackageInfo info = Api.getPackageDetails(context, listMaps, data.getUid());
+            if (info != null && info.applicationInfo != null) {
+                Drawable drawable = info.applicationInfo.loadIcon(manager);
+                holder.icon.setBackground(drawable);
+            } else {
                 Drawable appIcon = context.getDrawable(R.drawable.ic_unknown);
                 holder.icon.setImageBitmap(Api.getBitmapFromDrawable(appIcon));
             }
         } catch (Exception e) {
-            Log.e(TAG,e.getMessage(),e);
-            info = null;
+            Log.e(TAG, e.getMessage(), e);
             try {
-                if(Build.VERSION.SDK_INT > android.os.Build.VERSION_CODES.O) {
+                if (Build.VERSION.SDK_INT > android.os.Build.VERSION_CODES.O) {
                     holder.icon.setBackground(context.getDrawable(R.drawable.ic_unknown));
                 } else {
                     Drawable appIcon = context.getDrawable(R.drawable.ic_unknown);
                     holder.icon.setImageBitmap(Api.getBitmapFromDrawable(appIcon));
                 }
-            }catch (Exception e1) {
-                Log.e(TAG,e1.getMessage(),e1);
+            } catch (Exception e1) {
+                Log.e(TAG, e1.getMessage(), e1);
             }
         }
-
-
         try {
-            //if(data.getTimestamp() != null && !data.getTimestamp().isEmpty()) {
-            holder.lastDenied.setText(pretty(new Date(System.currentTimeMillis() - data.getTimestamp())));
-            //}
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTimeInMillis(data.getTimestamp());
+            DateFormat dateFormat = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.SHORT);
+            holder.lastDenied.setText(dateFormat.format(calendar.getTime()));
```

Comment:
This is the only place where that is used.

Reply:
Thanks for PR. Does it retain the same functionality as prettytime ?

Reply:
That's one of the java standard ways of printing a localized date/time (taking the time zone and ... into consideration).
It should do what's needed for the date/time string shown in the log view.

Reply:
Note: In the detail view of the log entries the date/time is shown without any localization.
Probably the localized time should be shown there too.

Reply:
I think the focus should be kept on fixing the crash first without adding complexity.
The pretty format can always be added on top...
I don't have much passion on which direction the solution goes, but I find it a pity having the app crash for something like that.
In my opinion as long as one can see the date/time of the logged event it's perfect (and it's even localized, so the US won't be mixing months with days ;)). What do you think?

Reply:
Agree with your point. But I don't see any crashing reports based on this.